client/src/components/ProfileSettingsSkeleton/ProfileSettingsSkeleton.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState, SyntheticEvent } from 'react';
+import { Link, Switch, Route } from 'react-router-dom';
+import { useHistory } from 'react-router';
+import clsx from 'clsx';
+import MenuIcon from '@material-ui/icons/Menu';
+import { Box, Grid, Typography, Button, Menu, MenuItem } from '@material-ui/core';
+import links from './ProfleSettingsLInks';
+import useStyles from './useStyles';
+
+export default function ProfileSkeleton(): JSX.Element {
+  const classes = useStyles();
+  const { location } = useHistory();
+  const [element, setElement] = useState<(EventTarget & Element) | null>(null);
+  const open = Boolean(element);
+
+  const handleClick = (event: SyntheticEvent) => {
+    setElement(event.currentTarget);
+  };
+  const handleClose = () => {
+    setElement(null);
+  };
```

Comment:
Crafty. My suggestion would be to use less-generic names (e.g. `handleMenuOpen/Close`).